### PR TITLE
awscli2: 2.1.17 -> 2.1.29

### PR DESCRIPTION
**Pull request by davegallant**
###### Motivation for this change

api changes and a bugfix

https://github.com/aws/aws-cli/blob/2.1.29/CHANGELOG.rst

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/admin/awscli2/default.nix [hunk 1]
```diff
@@ -26,16 +21,17 @@ let
...
   };
 
-in
-with py.pkgs; buildPythonApplication rec {
+in with py.pkgs;
+buildPythonApplication rec {
   pname = "awscli2";
-  version = "2.1.17"; # N.B: if you change this, change botocore to a matching version too
+  version =
+    "2.1.29"; # N.B: if you change this, change botocore to a matching version too
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/awscli2/default.nix:28]:
```suggestion
  version = "2.1.29"; # N.B: if you change this, change botocore to a matching version too
```

File: pkgs/tools/admin/awscli2/default.nix [hunk 2]
```diff
@@ -83,7 +79,8 @@ with py.pkgs; buildPythonApplication rec {
   passthru.python = py; # for aws_shell
 
   meta = with lib; {
-    homepage = "https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html";
+    homepage =
+      "https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html";
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/awscli2/default.nix:83]:
```suggestion
    homepage = "https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html";
```

File: pkgs/tools/admin/awscli2/default.nix [hunk 3]
```diff
@@ -26,16 +21,17 @@ let
     };
   };
 
-in
-with py.pkgs; buildPythonApplication rec {
+in with py.pkgs;
+buildPythonApplication rec {
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/awscli2/default.nix:25]:
```suggestion
in
with py.pkgs; buildPythonApplication rec {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please use nixpkgs-fmt which does not enforce short lines.

**Comment by davegallant** [comment]
> Please use nixpkgs-fmt which does not enforce short lines.

Fixed. My apologies. My editor had `nixfmt` on save.

File: pkgs/tools/admin/awscli2/default.nix [hunk 4]
```diff
@@ -29,13 +24,13 @@ let
 in
 with py.pkgs; buildPythonApplication rec {
   pname = "awscli2";
-  version = "2.1.17"; # N.B: if you change this, change botocore to a matching version too
+  version = "2.1.29"; # N.B: if you change this, change botocore to a matching version too
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/awscli2/default.nix:27]: Does botocore require an update?

Comment (davegallant) [reply pkgs/tools/admin/awscli2/default.nix:27]: Yeah, it's being bumped in this PR:
https://github.com/NixOS/nixpkgs/blob/6f4b61d2ce39057fc0d9971fa7620cc7d7e7dd3a/pkgs/tools/admin/awscli2/default.nix#L6

**Pull request merged**